From: Jens Axboe <axboe@fb.com>
Date: Wed, 24 Aug 2016 15:52:48 -0600
Subject: block: add kblockd_schedule_work_on()
References: bnc#1023798,FATE#321463
Patch-Mainline: v4.9-rc1
Git-commit: ee63cfa7fc197b63669623721b8009cce5b0659b

Add a helper to schedule a regular struct work on a particular CPU.

Signed-off-by: Jens Axboe <axboe@fb.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 block/blk-core.c       | 6 ++++++
 include/linux/blkdev.h | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/block/blk-core.c b/block/blk-core.c
index 3064c3b..a5f86dc 100644
--- a/block/blk-core.c
+++ b/block/blk-core.c
@@ -3092,6 +3092,12 @@ int kblockd_schedule_work(struct work_struct *work)
 }
 EXPORT_SYMBOL(kblockd_schedule_work);
 
+int kblockd_schedule_work_on(int cpu, struct work_struct *work)
+{
+	return queue_work_on(cpu, kblockd_workqueue, work);
+}
+EXPORT_SYMBOL(kblockd_schedule_work_on);
+
 int kblockd_schedule_delayed_work(struct delayed_work *dwork,
 				  unsigned long delay)
 {
diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index 1bda81b..2e14681 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -1429,8 +1429,8 @@ static inline bool req_gap_front_merge(struct request *req, struct bio *bio)
 	return bio_will_gap(req->q, bio, req->bio);
 }
 
-struct work_struct;
 int kblockd_schedule_work(struct work_struct *work);
+int kblockd_schedule_work_on(int cpu, struct work_struct *work);
 int kblockd_schedule_delayed_work(struct delayed_work *dwork, unsigned long delay);
 int kblockd_schedule_delayed_work_on(int cpu, struct delayed_work *dwork, unsigned long delay);
 
-- 
1.8.5.6

